packages/@aws-cdk/aws-lambda/lib/function-url.ts

```diff
@@ -0,0 +1,213 @@
+import * as iam from '@aws-cdk/aws-iam';
+import { Duration, IResource, Resource } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { IAlias } from './alias';
+import { IFunction } from './function-base';
+import { IVersion } from './lambda-version';
+import { CfnUrl } from './lambda.generated';
+
+
+/**
+ * The auth types for a function url
+ */
+export enum FunctionUrlAuthType {
+  /**
+   * Restrict access to authenticated IAM users only
+   */
+  AWS_IAM = 'AWS_IAM',
+
+  /**
+   * Bypass IAM authentication to create a public endpoint
+   */
+  NONE = 'NONE',
+}
+
+/**
+ * All http request methods
+ */
+export enum HttpMethods {
```

Comment:
Any reason why `OPTIONS` is not present in this enum?

Reply:
Thanks for pointing out. I have added `PATCH` and `OPTIONS` along with `ALL` for the wildcard `*`